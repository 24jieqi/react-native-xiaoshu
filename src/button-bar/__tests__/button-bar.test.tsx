import React from 'react'
import { SafeAreaProvider } from 'react-native-safe-area-context'
import { create } from 'react-test-renderer'

import ButtonBar from '../'

describe('ButtonBar', () => {
  it('render snapshot', () => {
    const tree = create(
      <SafeAreaProvider>
        <ButtonBar
          accessibilityLabel="ButtonBar"
          buttons={[
            {
              text: '文案',
            },
            {
              text: '操作',
            },
          ]}
        />
      </SafeAreaProvider>,
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
