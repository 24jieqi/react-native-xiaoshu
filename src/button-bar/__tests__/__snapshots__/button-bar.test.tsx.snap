// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonBar render snapshot 1`] = `
<View
  accessibilityLabel="ButtonBar"
  collapsable={false}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderTopColor": "#EFF3F9",
      "borderTopWidth": 1,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "flex-end",
      "overflow": "hidden",
      "paddingBottom": 0,
      "paddingHorizontal": 12,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "justifyContent": "flex-end",
          "paddingLeft": 0,
          "paddingRight": 0,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "marginBottom": 0,
            "marginRight": 8,
            "minWidth": undefined,
          },
          null,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#0065FE",
            "borderColor": "#0065FE",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "flexDirection": "column",
            "height": 40,
            "justifyContent": "center",
            "minWidth": 92,
            "opacity": 1,
            "paddingHorizontal": 8,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "flexWrap": "nowrap",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              {
                "color": "#FFFFFF",
                "fontSize": 15,
                "marginLeft": 0,
                "textAlign": "center",
              }
            }
          >
            操作
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "marginBottom": 0,
            "marginRight": 8,
            "minWidth": undefined,
          },
          {
            "marginRight": 0,
          },
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#0065FE",
            "borderColor": "#0065FE",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "flexDirection": "column",
            "height": 40,
            "justifyContent": "center",
            "minWidth": 92,
            "opacity": 1,
            "paddingHorizontal": 8,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "flexWrap": "nowrap",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              {
                "color": "#FFFFFF",
                "fontSize": 15,
                "marginLeft": 0,
                "textAlign": "center",
              }
            }
          >
            文案
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
